OutOfSchool/OutOfSchool.WebApi.Tests/Services/ProviderServicesTests/ProviderTestsHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Moq;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+
+namespace OutOfSchool.WebApi.Tests.Services.ProviderServicesTests;
+
+public static class ProviderTestsHelper
+{
+    public static Mock<IProviderRepository> CreateProvidersRepositoryMock(IEnumerable<Provider> providersCollection)
+    {
+        var providersRepository = new Mock<IProviderRepository>();
+        var userExistsResult = false;
+
+        bool UserExist(string userId)
+        {
+            userExistsResult = providersCollection.Any(p => p.UserId.Equals(userId));
+            return userExistsResult;
+        }
```

Comment:
(Readability) Taking into account comment below, you could get rid of `userExistsResult` and `UserExist()` method and inline it in mock `.Returns` lambda.